### ISSUE TITLE
About training time

### ISSUE THREAD

**Issue by BoPang1996** [valeoai/SLidR#6]
Hi, thanks for the code. This is not an issue. 

When I pretrain the Minkunet with the provided config, it takes 4 hours to train one epoch on 1 V100 GPU. That is 200 hours for pretraining. Is this right?  I notice that the code supports ddp training. Is it OK to train with more gpus? Will it cause a performance drop?

**Comment by CSautier** [comment]
Hi,
Your pretraining time is much higher than it should be on a V100. Most likely it comes from the dataloader or MinkowskiEngine not using all your cpu cores correctly. 
In the configs there is a parameter "num_threads", you should adapt it to your number of cores to speedup the dataloader.
As for MinkowskiEngine, you should use `export OMP_NUM_THREADS=n` with n being your number of cores.
You can also use multi-gpu, it is supported and does not affect results too much, we have obtained similar results using a pre-training with 4 and 8 GPUs.

**Comment by BoPang1996** [comment]
Thanks for the reply. I have enlarged the num_threads for loader, and I will try the OMP_NUM_THREADS. BTW, what is the expected pre-training time on 1GPU? If I adopt 8 GPUs, is it just simply adopt an 8x batchsize, 8x lr to train? Is there any other hyper-parameter to modify?

**Comment by BoPang1996** [comment]
My CPU got 22 cores and 44 threads. When I adopt num_threads=16 and OMP_NUM_THREADS=24, one epoch on 1 GPU takes about 2.25 hours.

**Comment by CSautier** [comment]
This is still a bit slow, but at least there is progress. An epoch on V100S took about 1h30, and I used fewer cores than that. I cannot tell you exactly what slows your experiment down, but MinkowskiEngine's performance seems to vary a lot depending on the CUDA version and the GPU used.
If you decide to do multi-GPU, please scale the learning rate and batch size accordingly (simply multiply the batch size and learning rate by the number of GPUs).

**Comment by Jeonghoon4** [comment]
I had a same problem.
I solve it to set num_threads=4, OMP_NUM_THREADS=4 with a single 3090 gpu.
one epoch take about 1.5 hours

For single 3090 gpu,
num_threads=4, OMP_NUM_THREADS=16 -> 2.5 hours / epoch
num_threads=4, OMP_NUM_THREADS=4 -> 1.5 hours / epoch
